tests/utils.go

```diff
@@ -2092,17 +2092,18 @@ func NewRandomFedoraVMIWithDmidecode() *v1.VirtualMachineInstance {
 }
 
 func GetGuestAgentUserData() string {
+	guestAgentUrl := GetUrl(GuestAgentHttpUrl)
 	return fmt.Sprintf(`#!/bin/bash
                 echo "fedora" |passwd fedora --stdin
                 mkdir -p /usr/local/bin
-                sleep 1
+                for i in {1..20}; do curl -I %s | grep "200 OK" && break || sleep 0.1; done
```

Comment:
NetworkManager is being started before cloud-init, and abducts the cloud init interface when it created.

The `NetworkManager-wait-online.service` service ends up waiting 30 seconds for NetworkManager to finish configuring the interface. Only once this fails can cloud-init proceed to userdata.

This is caused by https://github.com/canonical/cloud-init/commit/8888ca1af7f1fed33e79b56694834310ed35559a, which was introduced in an attempt to fix https://bugs.launchpad.net/cloud-init/+bug/1843334.